test/e2e/util.go

```diff
@@ -2098,7 +2098,7 @@ func NewHostExecPodSpec(ns, name string) *api.Pod {
 	pod := &api.Pod{
 		TypeMeta: unversioned.TypeMeta{
 			Kind:       "Pod",
-			APIVersion: latest.GroupOrDie("").GroupVersion.Version,
```

Comment:
> The new code asks for some preferred group's version. The old code is much preferable because it's explicit what group the code is working with.

I can refactor the `GroupMeta` to have a better named fields and make the call use a proper constant, so it becomes `latest.GroupOrDie(api.SchemeGroupVersion.Group).PreferredExternalVersion.String()` as opposed to `latest.PreferredExternalVersion.String()`.

@lavalamp I can do that refactor first to make this one more obvious, but basically this is already doing something that doesn't make sense.